ninja-servlet/src/main/java/ninja/servlet/NinjaBootstrap.java

```diff
@@ -191,7 +206,15 @@ protected void configureServlets() {
             return injector;
 
         } catch (Exception exception) {
-            logger.error("Fatal error booting Ninja", exception);
+            
+            // save exception in case other consumers of this class would like
+            // to handle an injector exception a little differently
+            injectorException = exception;
```

Comment:
So, I dislike how I implemented this as well, so its a good discussion.  The real problem is this line:

Preconditions.checkNotNull(injector, "Ninja injector cannot be generated. Please check log for further errors.");

The initInjector() method catches the real exception, logs it, then returns null.  Therefore the real exception causing a startup failure was hidden and merely logged.  Then the Preconditions.checkNotNull method is used to actually trigger the exception and your told "Please analyze log...".  NinjaJetty only got a RuntimeException of the "Please analyze log" exception.

I would actually have preferred to modify initInjector to throw an exception and not return null.  I just wasn't sure that was safe to do since it seems like initInjector was designed to NOT throw an exception.  So the safer route for me was to at least save the original exception and not log it -- so that it could be logged correctly in NinjaJetty.

BTW - Guice startup errors are one of the most annoying aspects of Guice (IMO).  You get 30+ errors and you really only care about the first few.  With the real exception in NinjaJetty, someone could eventually submit a PR to optimize for that error to help drill down quickly to the real problem.  That's why I cared about getting to the real exception.